### PR TITLE
tests build: buil on push/pr to master

### PR DESCRIPTION
**Pull request by pared**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Currently each time anyone pushes anything to any fork of DVC, the test build is triggered. I think we should not allow that. Since the only builds that are interesting to us are the ones for master commits/pull requests.

### CODE REVIEW

**Comment by efiop** [comment]
> Currently each time anyone pushes anything to any fork of DVC, the test build is triggered. I think we should not allow that. Since the only builds that are interesting to us are the ones for master commits/pull requests.

Are fork builds visible in our actions or only in fork's actions?

Currently, you only allow `master` to be built with our GHA, which is not the best idea, as it prevents people from using dev branches (e.g. we sometimes use those to backport something to an older release and release an old version e.g. 0.94). If fork's builds are accounted as our main repo builds then maybe you really meant something like `github.repository == 'iterative/dvc'`

**Comment by pared** [comment]
> Are fork builds visible in our actions or only in fork's actions?

Yes, fork builds are not visible in our repo.

> as it prevents people from using dev branches

It does not, user can just add the branch to "branches" and build will be done on this particular branch too.

> `github.repository == 'iterative/dvc'`

This is also a good idea,
I think we should combine both conditions - in current state builds should be happening on PR's and pushes to master, only in our repository, expanding permissions to any other situation is a bit wasteful in my opinion.

**Review by efiop — APPROVED** [review]

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
 
 jobs:
   lint:
+    if: ${{ github.repository }} == 'iteratie/dvc'
```
Comment (skshetry) [inline .github/workflows/tests.yaml:13]: I don't quite understand why this change was necessary, as the workflow for forks run on their own repository.
There is a spelling error too. `s/iteratie/iterative`.

Comment (pared) [reply .github/workflows/tests.yaml:13]: So, since we promote PR's from users repositories, it is just wasteful to run full builds. Typo is a good point, and since the recent builds are still building, it seems like its not respected. Will get rid of that.
